frontend/form-builder/src/components/ComponentSettings/ButtonGroup.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useContext } from "react";
+import classes from "./ComponentSettings.module.css";
+import { Button } from "@digdir/design-system-react";
+import { FloppydiskFillIcon, TrashFillIcon } from "@navikt/aksel-icons";
+import { FormBuilderContext } from "../../context";
+import { validateOrder } from "../../../../main-app/src/components/SubHeader/utils";
+import { useTranslation } from "react-i18next";
+
+export const ButtonGroup = () => {
+  const { t } = useTranslation();
+  const { form, setForm, currentComponent, setCurrentComponent } = useContext(FormBuilderContext);
+
+  const handleDeleteComponent = (index: number) => {
+    const newForm = form;
+    newForm.components = form.components.filter((_, i) => i !== index);
+    validateOrder(newForm);
+
+    setForm({ ...form, ...newForm });
```

Comment:
and then we update `setForm `:
`   setForm(prevForm => ({...prevForm, ...newForm}))`

Reply:
Do we need to use prevForm here? How about this?
@lassopicasso 
```
const handleDeleteComponent = (index: number) => {
    const newForm: Form = { ...form };
    newForm.components = form.components.filter((_, i) => i !== index);
    validateOrder(newForm);

    setForm({ ...newForm });
    setCurrentComponent(null);
  };
```

Reply:
I think `setForm({ ...newForm });` works fine in this case. Since there are no other state changes above in the `handleDeleteComponent` function.

The `newForm` could also be set like this;
```
const newForm: Form = {
  ...form,
  components: form.components.filter((_, i) => i !== indexToRemove)
};
```

Reply:
Smart :)